prover/src/channel.rs

```diff
@@ -137,18 +137,18 @@ where
     /// constraint composition polynomials should be queried.
     ///
     /// The positions are drawn from the public coin uniformly at random.
-    pub fn get_query_positions(&mut self) -> Vec<usize> {
+    pub fn get_query_positions(&mut self, nonce: u64) -> Vec<usize> {
         let num_queries = self.context.options().num_queries();
         let lde_domain_size = self.context.lde_domain_size();
         self.public_coin
-            .draw_integers(num_queries, lde_domain_size)
+            .draw_integers(num_queries, lde_domain_size, nonce)
             .expect("failed to draw query position")
     }
 
     /// Determines a nonce, which when hashed with the current seed of the public coin results
     /// in a new seed with the number of leading zeros equal to the grinding_factor specified
     /// in the proof options.
-    pub fn grind_query_seed(&mut self) {
+    pub fn grind_query_seed(&mut self) -> u64 {
```

Comment:
If the above comment applies, then we can probably leave this method as it was before.